parcels/field.py

```diff
@@ -600,17 +600,21 @@ def set_scaling_factor(self, factor):
         self._scaling_factor = factor
         self.data *= factor
 
-    def _search_indices(self, time, z, y, x, ti, particle=None, search2D=False):
+    def _search_indices(self, time, z, y, x, particle=None, search2D=False):
+        tau, ti = self._time_index(time)
+
         if self.grid._gtype in [GridType.RectilinearSGrid, GridType.RectilinearZGrid]:
-            return _search_indices_rectilinear(self, time, z, y, x, ti, particle=particle, search2D=search2D)
+            (zeta, eta, xsi, zi, yi, xi) = _search_indices_rectilinear(
+                self, time, z, y, x, ti, particle=particle, search2D=search2D
+            )
         else:
-            return _search_indices_curvilinear(self, time, z, y, x, ti, particle=particle, search2D=search2D)
+            (zeta, eta, xsi, zi, yi, xi) = _search_indices_curvilinear(
+                self, time, z, y, x, ti, particle=particle, search2D=search2D
+            )
+        return (tau, zeta, eta, xsi, ti, zi, yi, xi)
```

Comment:
I think


```python
        if self.grid._gtype in [GridType.RectilinearSGrid, GridType.RectilinearZGrid]:
            spatial_component = _search_indices_rectilinear(
                self, time, z, y, x, ti, particle=particle, search2D=search2D
            )
        else:
            spatial_component = _search_indices_curvilinear(
                self, time, z, y, x, ti, particle=particle, search2D=search2D
            )
        return tau, *spatial_component
```
Would be a bit less repetitive? Though users would need to look at `_search_indices_*` to understand what's actually being returned, which I think is fine.

Reply:
Hmm, note that it also injects `ti` halfway, so I'm not in favour of this. I'd prefer the clarity over the reduced repetitiveness...

Reply:
Ah! Didn't see that :)